Sources/NIOExtras/LengthFieldBasedFrameDecoder.swift

```diff
@@ -17,6 +17,8 @@ import NIO
 public enum LengthFieldBasedFrameDecoderError: Error {
     /// This error can be thrown by `LengthFieldBasedFrameDecoder` if the length field value is larger than `Int.max`
     case lengthFieldValueTooLarge
+    /// This error can be thrown by `LengthFieldBasedFrameDecoder` if the length field value is larger than `LengthFieldBasedFrameDecoder.maxSupportedLengthFieldSize`
+    case lengthFieldValueLargerThanMaxSupportedSize
```

Comment:
this is also public API breaking 😭. You have two options:

1. just use `lengthFieldValueTooLarge` (which is what I would do)
2. create a new type inside `NIOExtraError.swift` called `LengthFieldValueLargerThanMaximumSupported` or similar

Reply:
The enum is completely new in this pull request. The case was just added later. But if you like, I can still remove the separate enum case :)
BTW: Is there a recommended way of defining error's? If I remember correctly, only libraries which define ABI stability can add new enum cases to enum's which are not marked as `@forzen` but there was proposal to fix that kind of issue in non ABI stable libraries but wasn't accepted.

Reply:
@dnadoba ah, that's great. In this case you just need to prefix the `enum` with `NIO` ie `NIOLengthFieldBased...` so we don't violate our [public API guarantees](https://github.com/apple/swift-nio/blob/main/docs/public-api.md#1-no-global-namespace-additions-that-arent-prefixed).

You are correct, that SwiftPM libraries don't have the option between "open" and `@frozen` enums sadly. All we've got are `@frozen` ones (without specifying that).

I'm happy with adding this `enum` however we could also move both into `NIOExtrasError`. Let's ask @Lukasa what he thinkgs